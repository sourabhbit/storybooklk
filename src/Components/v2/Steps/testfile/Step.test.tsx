import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import {
	ComponentSizeENUM,
	ThemeENUM,
	TypographyENUM,
} from '../../../../Types/general'
import Step from '../index'

const props = {
	id: 'step',
	totalStep: 2,
	currentStep: 1,
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.primary,
	font: TypographyENUM.serif,
	onClick: () => null,
}

describe('<Step />', () => {
	let _asFragment: any = null

	beforeEach(() => {
		const { asFragment } = render(<Step {...props} />)
		_asFragment = asFragment
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('step-wrapper')
		expect(container).toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
