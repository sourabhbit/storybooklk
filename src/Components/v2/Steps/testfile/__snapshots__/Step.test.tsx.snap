// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> Should take a snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-bdvvtL bMHWKf"
    data-testid="step-wrapper"
    id="step"
  >
    <div
      class="sc-gsDKAQ loNbFt"
    >
      <img
        alt="back"
      />
    </div>
    <span>
      Step 1/2
    </span>
  </span>
</DocumentFragment>
`;
